### PR TITLE
blukon: raw command strings converted to readable names

### PR DESCRIPTION
**Pull request by keencave**
added additional comments how the communication works.

It should be easier now to understand and/or change the java code for blucon. The raw strings were replaced by static String variables. Comments added where to find which byte in which block of the FRAM.

The code compiles without warnings. I made a line by line check but are not able to real test it within the next days.

### CODE REVIEW

**Comment by gregorybel** [comment]
Great work!
Will test it if you want.
but from review, I don't see problems.

**Comment by keencave** [comment]
yes, please test it if you have time. I made the changes carefully but we better test it.

File: app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java [hunk 1]
```diff
@@ -256,58 +341,67 @@ public static void unBondIfBlukonAtInit() {
...
+                currentCommand = SLEEP_COMMAND;
                 Log.i(TAG, "Send sleep cmd");
                 m_communicationStarted = false;
             }
 
-        } else if (currentCommand.startsWith("010d0b00") /*getUnknownCmd1*/ && strRecCmd.startsWith("8bdb")) {
+        /*
+         * step 5: send unknownCommand1 as otherwise communication errors will occur
+         */
+        } else if (currentCommand.startsWith(UNKNOWN1_COMMAND) /*getUnknownCmd1*/ && strRecCmd.startsWith("8bdb")) {
```
Comment (dabear) [inline app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java]: Convert this to a readable name as well?

File: app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java [hunk 2]
```diff
@@ -256,58 +341,67 @@ public static void unBondIfBlukonAtInit() {
...
 
-            currentCommand = "010d0a00";
+            currentCommand = UNKNOWN2_COMMAND;
             Log.i(TAG, "getUnknownCmd2 " + currentCommand);
 
-        } else if (currentCommand.startsWith("010d0a00") /*getUnknownCmd2*/ && strRecCmd.startsWith("8bda")) {
+        /*
+         * step 6: send unknownCommand2 as otherwise communication errors will occur
+         */
+        } else if (currentCommand.startsWith(UNKNOWN2_COMMAND) /*getUnknownCmd2*/ && strRecCmd.startsWith("8bda")) {
```
Comment (dabear) [inline app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java]: Here as well?

**Comment by gregorybel** [comment]
I tested it and it works well. I think we can merge.

**Comment by jamorham** [comment]
Is everyone happy with this? It looks good to me although I haven't verified everything has been transcribed safely. Good improvement to reliability.

**Comment by gregorybel** [comment]
yes even if @dabear is right, this could be improved too

**Comment by keencave** [comment]
as soon we added trallen's PR I will add the missing commands. Thanks dabear!

**Pull request merged**